src/parser/mod.rs

```diff
@@ -532,6 +532,14 @@ impl<'a> Parser<'a> {
                 Keyword::EXECUTE => self.parse_execute(),
                 Keyword::PREPARE => self.parse_prepare(),
                 Keyword::MERGE => self.parse_merge(),
+                // `LISTEN` and `NOTIFY` are Postgres-specific
+                // syntaxes. They are used for Postgres statement.
+                Keyword::LISTEN if dialect_of!(self is PostgreSqlDialect | GenericDialect) => {
+                    self.parse_listen()
+                }
+                Keyword::NOTIFY if dialect_of!(self is PostgreSqlDialect | GenericDialect) => {
+                    self.parse_notify()
+                }
```

Comment:
Can we use a dialect method `dialect.supports_listen_notify()` to guard the behavior?

Reply:
Good idea! Have Done!